ebean-core/src/main/java/io/ebeaninternal/server/query/DefaultOrmQueryEngine.java

```diff
@@ -133,7 +133,7 @@ public boolean isMultiValueSupported(Class<?> cls) {
 
     SpiQuery<T> query = request.getQuery();
 
-    if (request.isBeanCachePutMany()) {
+    if (result != null && request.isBeanCachePutMany()) {
```

Comment:
Slightly odd case where we query loading existing beans and intentionally returning a null result. Odd but ok for now.